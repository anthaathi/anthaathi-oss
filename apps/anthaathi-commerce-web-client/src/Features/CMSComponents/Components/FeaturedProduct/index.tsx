import { useStyletron } from '@anthaathi/solid-styletron';
import {
  IconDotCircleOLarge,
  IconGlobeLarge,
  IconLeafLarge,
  IconLockLarge,
} from '@anthaathi/oracle-apex-solid-icons';
import { FAQ } from '../FAQ';
import { Img } from '~/Features/Core/Components/Image';
import { Button } from 'solid-headless';

type BlockInfoProps = {
  freeShipping: string;
  isFresh: string;
  securePayments: string;
  inStock: string;
};

type ListInfoProps = {
  description: string;
  shippingInformation: string;
};
export interface ProductDetails {
  name: string;
  listInfo?: ListInfoProps;
  blockInfo: BlockInfoProps;
  image: string[];
  price: number;
  currency: string;
}

export interface ProductDetailsProps {
  productInfo: ProductDetails;
  handleAddToCart?: () => void;
  handleBuyItNow?: () => void;
}

export function FeaturedProduct(props: ProductDetailsProps) {
  const [css, $theme] = useStyletron();
  return (
    <div
      class={css({
        marginLeft: 'auto',
        marginRight: 'auto',
        width: $theme.sizing.maxWidth,
        maxWidth: `calc(100% - ${$theme.sizing.scale500} - ${$theme.sizing.scale500})`,
        paddingLeft: $theme.sizing.scale500,
        paddingRight: $theme.sizing.scale500,
        marginTop: $theme.sizing.scale1000,
        marginBottom: $theme.sizing.scale1000,
      })}
    >
      <div
        class={css({
          marginBottom: $theme.sizing.scale1000,
          display: 'flex',
          flexDirection: 'column',
          [$theme.mediaQuery?.md || '']: {
            flexDirection: 'row',
          },
        })}
      >
        <div
          class={css({
            flex: 1,
            alignItems: 'center',
            display: 'flex',
            placeContent: 'center',
          })}
        >
          <Img
            src={props.productInfo.image[0]}
            $override={{
              Root: {
                $style: {
                  height: '520px',
                  width: '100%',
                  objectFit: 'cover',
                  [$theme.mediaQuery?.xl || '']: {
                    height: '520px',
                  },
                  [$theme.mediaQuery?.lg || '']: {
                    height: '420px',
                  },
                  [$theme.mediaQuery?.md || '']: {
                    width: '90%',
                  },
                  [$theme.mediaQuery?.sm || '']: {
                    height: '420px',
                  },
                  [$theme.mediaQuery?.xs || '']: {
                    height: '320px',
                  },
                },
              },
            }}
          />
        </div>

        <div
          class={css({
            flex: 1,
            //   backgroundColor: 'red',
          })}
        >
          <p
            class={css({
              ...$theme.typography.HeadingLarge,
              marginTop: $theme.sizing.scale600,
              marginBottom: $theme.sizing.scale600,
              fontWeight: 'bold',
              color: '#000',
            })}
          >
            {props.productInfo.name}
          </p>
          <p
            class={css({
              ...$theme.typography.LabelLarge,
              marginTop: $theme.sizing.scale600,
              marginBottom: 0,
              fontWeight: 'bold',
              color: '#000',
            })}
          >
            Price
          </p>

          <p
            class={css({
              ...$theme.typography.LabelLarge,
              marginTop: $theme.sizing.scale200,
              marginBottom: $theme.sizing.scale600,
              fontWeight: 'bold',
              color: '#000',
            })}
          >
            {Intl.NumberFormat('en-US', {
              style: 'currency',
              currency: props.productInfo.currency ?? 'AED',
            }).format(props.productInfo.price)}
          </p>
          <BlockInfo data={props.productInfo.blockInfo} />
          <Button
            onClick={props.handleAddToCart}
            class={css({
              textAlign: 'center',
              marginTop: $theme.sizing.scale600,
              width: '100%',
              [$theme.mediaQuery?.md || '']: {
                width: '80%',
              },
              paddingTop: '12px',
              paddingBottom: '12px',
              fontWeight: 'bold',
              fontSize: '18px',
              outline: 'none',
              border: 'none',
              borderRadius: '2px',
              color: '#313652',
              marginVertical: 10,
              ':hover': { cursor: 'pointer' },
            })}
          >
            Add to cart
          </Button>
          <div
            onClick={props.handleBuyItNow}
            class={css({
              marginTop: '10px',
              width: '100%',
              [$theme.mediaQuery?.md || '']: {
                width: '80%',
              },
              textAlign: 'center',
              border: '1px solid #313652',
              backgroundColor: '#313652',
              paddingTop: '12px',
              paddingBottom: '12px',
              color: '#fff',
              fontWeight: 'bold',
              fontSize: '18px',
              borderRadius: '2px',
              ':hover': { cursor: 'pointer' },
            })}
          >
            Buy it now
          </div>
        </div>
      </div>
      <FAQ
        list={[
          {
            question: 'Description',
            answer: `
              100% fresh.
              Sourced from India.
              Benefits:
              Mango with its rich flavor and amazing taste is a perfect refreshing beverage for scorching summer months. It is low in calories and a source of vital nutrients comprising dietary fiber, vitamins, and minerals.
              `,
          },
          {
            question: 'Shipping Information',
            answer: `
              Use collapsible tabs for more detailed information that will help customers make a purchasing decision.
              Ex: Shipping and return policies, size guides, and other common questions.
              `,
          },
        ]}
      />
    </div>
  );
}

const BlockInfo = ({ data }: { data: BlockInfoProps }) => {
  const [css, $theme] = useStyletron();
  return (
    <>
      <TextIcon
        title={data.freeShipping}
        icon={<IconGlobeLarge height="20px" width="20px" />}
      />
      <TextIcon
        title={data.isFresh}
        icon={<IconLeafLarge height="20px" width="20px" />}
      />
      <TextIcon
        title={data.securePayments}
        icon={<IconLockLarge height="20px" width="20px" />}
      />
      <TextIcon
        title={data.inStock}
        icon={
          <IconDotCircleOLarge color="#54c63a" height="20px" width="20px" />
        }
      />
    </>
  );
};

const TextIcon = ({ icon, title }: { icon: any; title: string }) => {
  const [css, $theme] = useStyletron();
  return (
    <div
      class={css({
        display: 'flex',
        flexDirection: 'row',
        alignItems: 'center',
        marginTop: $theme.sizing.scale300,
        marginBottom: $theme.sizing.scale300,
      })}
    >
      {icon}
      <p
        class={css({
          ...$theme.typography.ParagraphSmall,
          marginTop: 0,
          marginBottom: 0,
          marginLeft: $theme.sizing.scale300,
          color: '#000',
        })}
      >
        {title}
      </p>
    </div>
  );
};
