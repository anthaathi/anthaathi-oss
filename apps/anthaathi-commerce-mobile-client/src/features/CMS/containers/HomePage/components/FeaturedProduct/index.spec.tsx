import React from 'react';
import 'react-native';
import {fireEvent, render} from '@testing-library/react-native';
import FeaturedProduct from './index';
import {IntlProvider} from 'react-intl';

import locale from '../../../../../../compiled-locales/en-US.json';
import {ThemeProvider} from 'react-native-paper';
import {RecoilRoot} from 'recoil';

describe('FeaturedProduct', () => {
  it('should render the item', function () {
    const temp = render(
      <ThemeProvider>
        <RecoilRoot>
          <IntlProvider locale="en-US" messages={locale}>
            <FeaturedProduct
              productInfo={{
                name: 'test',
                listInfo: {
                  description: 'test',
                  shippingInformation: 'Shipping Information',
                },
                blockInfo: {
                  freeShipping: 'Free shipping in UAE',
                  inStock: 'In stock, ready to ship',
                  securePayments: 'Secure Payments',
                  isFresh: 'Fresh',
                },
                price: 0,
                currency: 'USD',
                image: [
                  'https://burst.shopifycdn.com/photos/tea-cup-with-hot-peppers-and-yellow-tomatoes-on-red.jpg?width=240&format=pjpg&exif=1&iptc=1',
                ],
              }}
            />
          </IntlProvider>
        </RecoilRoot>
      </ThemeProvider>,
    );

    expect(temp).toMatchSnapshot();
    expect(temp.queryByTestId('featuredProduct')).toBeTruthy();
  });

  it('should have FeaturedProduct name, description', () => {
    const temp = render(
      <ThemeProvider>
        <RecoilRoot>
          <IntlProvider locale="en-US" messages={locale}>
            <FeaturedProduct
              productInfo={{
                name: 'Name test',
                listInfo: {
                  description: 'test',
                  shippingInformation: 'Shipping Information',
                },
                blockInfo: {
                  freeShipping: 'Free shipping in UAE',
                  inStock: 'In stock, ready to ship',
                  securePayments: 'Secure Payments',
                  isFresh: 'Fresh',
                },
                price: 0,
                currency: 'USD',
                image: [
                  'https://burst.shopifycdn.com/photos/tea-cup-with-hot-peppers-and-yellow-tomatoes-on-red.jpg?width=240&format=pjpg&exif=1&iptc=1',
                ],
              }}
            />
          </IntlProvider>
        </RecoilRoot>
      </ThemeProvider>,
    );

    expect(temp.queryByText('Name test')).toBeTruthy();
  });

  
  it('should have FeaturedProduct name', () => {
    const temp = render(
      <ThemeProvider>
        <RecoilRoot>
          <IntlProvider locale="en-US" messages={locale}>
            <FeaturedProduct
              productInfo={{
                name: 'test',
                listInfo: {
                  description: 'test',
                  shippingInformation: 'Shipping Information',
                },
                blockInfo: {
                  freeShipping: 'Free shipping in UAE',
                  inStock: 'In stock, ready to ship',
                  securePayments: 'Secure Payments',
                  isFresh: 'Fresh',
                },
                price: 0,
                currency: 'USD',
                image: [
                  'https://burst.shopifycdn.com/photos/tea-cup-with-hot-peppers-and-yellow-tomatoes-on-red.jpg?width=240&format=pjpg&exif=1&iptc=1',
                ],
              }}
            />
          </IntlProvider>
        </RecoilRoot>
      </ThemeProvider>,
    );

    expect(temp.queryByText('test')).toBeTruthy();
  });

  it('should have FeaturedProduct Image', () => {
    const temp = render(
      <ThemeProvider>
        <RecoilRoot>
          <IntlProvider locale="en-US" messages={locale}>
            <FeaturedProduct
              productInfo={{
                name: 'test',
                listInfo: {
                  description: 'test',
                  shippingInformation: 'Shipping Information',
                },
                blockInfo: {
                  freeShipping: 'Free shipping in UAE',
                  inStock: 'In stock, ready to ship',
                  securePayments: 'Secure Payments',
                  isFresh: 'Fresh',
                },
                price: 0,
                currency: 'USD',
                image: [
                  'https://burst.shopifycdn.com/photos/tea-cup-with-hot-peppers-and-yellow-tomatoes-on-red.jpg?width=240&format=pjpg&exif=1&iptc=1',
                ],
              }}
            />
          </IntlProvider>
        </RecoilRoot>
      </ThemeProvider>,
    );

    expect(temp.queryByTestId('productImage')!.props.source).toMatchObject({
      uri: 'https://burst.shopifycdn.com/photos/tea-cup-with-hot-peppers-and-yellow-tomatoes-on-red.jpg?width=240&format=pjpg&exif=1&iptc=1',
    });
  });

  it('should have button title', function () {
    const temp = render(
      <ThemeProvider>
        <RecoilRoot>
          <IntlProvider locale="en-US" messages={locale}>
            <FeaturedProduct
              productInfo={{
                name: 'test',
                listInfo: {
                  description: 'test',
                  shippingInformation: 'Shipping Information',
                },
                blockInfo: {
                  freeShipping: 'Free shipping in UAE',
                  inStock: 'In stock, ready to ship',
                  securePayments: 'Secure Payments',
                  isFresh: 'Fresh',
                },
                price: 0,
                currency: 'USD',
                image: [
                  'https://burst.shopifycdn.com/photos/tea-cup-with-hot-peppers-and-yellow-tomatoes-on-red.jpg?width=240&format=pjpg&exif=1&iptc=1',
                ],
              }}
            />
          </IntlProvider>
        </RecoilRoot>
      </ThemeProvider>,
    );

    expect(temp.queryByText('Add to cart')).toBeTruthy();
    expect(temp.queryByText('Buy it now')).toBeTruthy();
  });

  it('should have number of badge for image selection', function () {
    const temp = render(
      <ThemeProvider>
        <RecoilRoot>
          <IntlProvider locale="en-US" messages={locale}>
            <FeaturedProduct
              productInfo={{
                name: 'test',
                listInfo: {
                  description: 'test',
                  shippingInformation: 'Shipping Information',
                },
                blockInfo: {
                  freeShipping: 'Free shipping in UAE',
                  inStock: 'In stock, ready to ship',
                  securePayments: 'Secure Payments',
                  isFresh: 'Fresh',
                },
                price: 0,
                currency: 'USD',
                image: [
                  'https://burst.shopifycdn.com/photos/tea-cup-with-hot-peppers-and-yellow-tomatoes-on-red.jpg?width=240&format=pjpg&exif=1&iptc=1',
                  'https://burst.shopifycdn.com/photos/tea-cup-with-hot-peppers-and-yellow-tomatoes-on-red.jpg?width=240&format=pjpg&exif=1&iptc=1',
                ],
              }}
            />
          </IntlProvider>
        </RecoilRoot>
      </ThemeProvider>,
    );

    expect(temp.queryAllByTestId('badgeSelectImage').length).toBe(2);
  });
});
