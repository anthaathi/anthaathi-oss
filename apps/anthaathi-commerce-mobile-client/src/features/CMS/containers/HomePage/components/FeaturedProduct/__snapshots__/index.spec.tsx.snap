// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FeaturedProduct should render the item 1`] = `
<View
  testID="featuredProduct"
>
  <View
    style={
      {
        "flexDirection": "row",
        "marginHorizontal": 12,
        "marginVertical": 10,
      }
    }
  >
    <View
      style={
        [
          {
            "alignItems": "center",
            "justifyContent": "center",
          },
          {
            "width": "50%",
          },
        ]
      }
    >
      <Image
        source={
          {
            "uri": "https://burst.shopifycdn.com/photos/tea-cup-with-hot-peppers-and-yellow-tomatoes-on-red.jpg?width=240&format=pjpg&exif=1&iptc=1",
          }
        }
        style={
          {
            "borderRadius": 4,
            "height": 280,
            "width": "70%",
          }
        }
        testID="productImage"
      />
      <View
        style={
          {
            "paddingVertical": 15,
          }
        }
      />
    </View>
    <View
      style={
        [
          {
            "width": "50%",
          },
        ]
      }
    >
      <Text
        style={
          [
            {
              "color": "#000000",
              "fontFamily": "System",
              "fontWeight": "400",
            },
            {
              "textAlign": "left",
            },
            {
              "fontSize": 20,
              "fontWeight": "600",
              "marginBottom": 9,
            },
          ]
        }
        variant="titleLarge"
      >
        test
      </Text>
      <Text
        style={
          [
            {
              "color": "#000000",
              "fontFamily": "System",
              "fontWeight": "400",
            },
            {
              "textAlign": "left",
            },
            {
              "fontSize": 16,
              "fontWeight": "500",
            },
          ]
        }
        variant="titleLarge"
      >
        Price
      </Text>
      <Text
        style={
          [
            {
              "color": "#000000",
              "fontFamily": "System",
              "fontWeight": "400",
            },
            {
              "textAlign": "left",
            },
            {
              "fontSize": 16,
              "fontWeight": "500",
              "marginBottom": 9,
            },
          ]
        }
        variant="titleLarge"
      >
        $0.00
      </Text>
      <View
        style={
          {
            "alignItems": "center",
            "flexDirection": "row",
            "marginVertical": 2,
          }
        }
      >
        <Text
          allowFontScaling={false}
          selectable={false}
          style={
            [
              {
                "color": "#000000",
                "fontSize": 16,
              },
              undefined,
              {
                "fontFamily": "Ionicons",
                "fontStyle": "normal",
                "fontWeight": "normal",
              },
              {},
            ]
          }
        >
          
        </Text>
        <Text
          style={
            [
              {
                "color": "#000000",
                "fontFamily": "System",
                "fontWeight": "400",
              },
              {
                "textAlign": "left",
              },
              {
                "fontSize": 14,
                "fontWeight": "400",
                "marginLeft": 10,
              },
            ]
          }
        >
          Free shipping in UAE
        </Text>
      </View>
      <View
        style={
          {
            "alignItems": "center",
            "flexDirection": "row",
            "marginVertical": 2,
          }
        }
      >
        <Text
          allowFontScaling={false}
          selectable={false}
          style={
            [
              {
                "color": "#000000",
                "fontSize": 16,
              },
              undefined,
              {
                "fontFamily": "Ionicons",
                "fontStyle": "normal",
                "fontWeight": "normal",
              },
              {},
            ]
          }
        >
          
        </Text>
        <Text
          style={
            [
              {
                "color": "#000000",
                "fontFamily": "System",
                "fontWeight": "400",
              },
              {
                "textAlign": "left",
              },
              {
                "fontSize": 14,
                "fontWeight": "400",
                "marginLeft": 10,
              },
            ]
          }
        >
          Fresh
        </Text>
      </View>
      <View
        style={
          {
            "alignItems": "center",
            "flexDirection": "row",
            "marginVertical": 2,
          }
        }
      >
        <Text
          allowFontScaling={false}
          selectable={false}
          style={
            [
              {
                "color": "#000000",
                "fontSize": 16,
              },
              undefined,
              {
                "fontFamily": "Ionicons",
                "fontStyle": "normal",
                "fontWeight": "normal",
              },
              {},
            ]
          }
        >
          
        </Text>
        <Text
          style={
            [
              {
                "color": "#000000",
                "fontFamily": "System",
                "fontWeight": "400",
              },
              {
                "textAlign": "left",
              },
              {
                "fontSize": 14,
                "fontWeight": "400",
                "marginLeft": 10,
              },
            ]
          }
        >
          Secure Payments
        </Text>
      </View>
      <View
        style={
          {
            "alignItems": "center",
            "flexDirection": "row",
            "marginVertical": 2,
          }
        }
      >
        <Text
          allowFontScaling={false}
          selectable={false}
          style={
            [
              {
                "color": "#000000",
                "fontSize": 16,
              },
              undefined,
              {
                "fontFamily": "Ionicons",
                "fontStyle": "normal",
                "fontWeight": "normal",
              },
              {},
            ]
          }
        >
          
        </Text>
        <Text
          style={
            [
              {
                "color": "#000000",
                "fontFamily": "System",
                "fontWeight": "400",
              },
              {
                "textAlign": "left",
              },
              {
                "fontSize": 14,
                "fontWeight": "400",
                "marginLeft": 10,
              },
            ]
          }
        >
          In stock, ready to ship
        </Text>
      </View>
      <View
        style={
          {
            "alignItems": "center",
            "backgroundColor": "#F1F9F4",
            "borderRadius": 50,
            "flexDirection": "row",
            "height": 35,
            "justifyContent": "space-between",
            "marginVertical": 10,
            "width": 110,
          }
        }
      >
        <View
          accessibilityRole="button"
          accessibilityState={
            {
              "disabled": true,
            }
          }
          accessible={true}
          focusable={false}
          hitSlop={
            {
              "bottom": 6,
              "left": 6,
              "right": 6,
              "top": 6,
            }
          }
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            [
              {
                "overflow": "hidden",
              },
              [
                {
                  "alignItems": "center",
                  "justifyContent": "center",
                  "margin": 6,
                  "overflow": "hidden",
                },
                {
                  "borderRadius": 15,
                  "height": 30,
                  "width": 30,
                },
                undefined,
                {
                  "backgroundColor": "#D4EDDD",
                  "borderRadius": 50,
                  "height": 35,
                  "marginLeft": 0,
                  "width": 35,
                },
              ],
            ]
          }
        >
          <View>
            <Text
              accessibilityElementsHidden={true}
              allowFontScaling={false}
              importantForAccessibility="no-hide-descendants"
              pointerEvents="none"
              selectable={false}
              style={
                [
                  {
                    "color": "#000000",
                    "fontSize": 20,
                  },
                  [
                    {
                      "lineHeight": 20,
                      "transform": [
                        {
                          "scaleX": 1,
                        },
                      ],
                    },
                    {
                      "backgroundColor": "transparent",
                    },
                  ],
                  {
                    "fontFamily": "Material Design Icons",
                    "fontStyle": "normal",
                    "fontWeight": "normal",
                  },
                  {},
                ]
              }
            >
              󰍴
            </Text>
          </View>
        </View>
        <Text
          style={
            [
              {
                "color": "#000000",
                "fontFamily": "System",
                "fontWeight": "400",
              },
              {
                "textAlign": "left",
              },
              {
                "color": undefined,
                "fontSize": 14,
                "fontWeight": "700",
                "marginHorizontal": 5,
                "textAlign": "center",
              },
            ]
          }
        >
          0
        </Text>
        <View
          accessibilityRole="button"
          accessibilityState={
            {
              "disabled": true,
            }
          }
          accessible={true}
          focusable={false}
          hitSlop={
            {
              "bottom": 6,
              "left": 6,
              "right": 6,
              "top": 6,
            }
          }
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            [
              {
                "overflow": "hidden",
              },
              [
                {
                  "alignItems": "center",
                  "justifyContent": "center",
                  "margin": 6,
                  "overflow": "hidden",
                },
                {
                  "borderRadius": 15,
                  "height": 30,
                  "width": 30,
                },
                undefined,
                {
                  "backgroundColor": "#D4EDDD",
                  "borderRadius": 50,
                  "height": 35,
                  "marginRight": 0,
                  "width": 35,
                },
              ],
            ]
          }
        >
          <View>
            <Text
              accessibilityElementsHidden={true}
              allowFontScaling={false}
              importantForAccessibility="no-hide-descendants"
              pointerEvents="none"
              selectable={false}
              style={
                [
                  {
                    "color": "#000000",
                    "fontSize": 20,
                  },
                  [
                    {
                      "lineHeight": 20,
                      "transform": [
                        {
                          "scaleX": 1,
                        },
                      ],
                    },
                    {
                      "backgroundColor": "transparent",
                    },
                  ],
                  {
                    "fontFamily": "Material Design Icons",
                    "fontStyle": "normal",
                    "fontWeight": "normal",
                  },
                  {},
                ]
              }
            >
              󰐕
            </Text>
          </View>
        </View>
      </View>
      <View
        collapsable={false}
        style={
          {
            "backgroundColor": "transparent",
            "borderColor": "#424242",
            "borderRadius": 1,
            "borderStyle": "solid",
            "borderWidth": 0.5,
            "elevation": 0,
            "marginVertical": 10,
            "minWidth": 64,
            "shadowColor": "#000000",
            "shadowOffset": {
              "height": 0,
              "width": 0,
            },
            "shadowOpacity": 0,
            "shadowRadius": 0,
          }
        }
      >
        <View
          accessibilityRole="button"
          accessibilityState={
            {
              "disabled": true,
            }
          }
          accessible={true}
          focusable={false}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            [
              {
                "overflow": "hidden",
              },
              {
                "borderRadius": 1,
              },
            ]
          }
          testID="handleAddToCart"
        >
          <View
            style={
              [
                {
                  "alignItems": "center",
                  "flexDirection": "row",
                  "justifyContent": "center",
                },
                undefined,
              ]
            }
          >
            <Text
              numberOfLines={1}
              selectable={false}
              style={
                [
                  {
                    "color": "#000000",
                    "fontFamily": "System",
                    "fontWeight": "400",
                  },
                  {
                    "textAlign": "left",
                  },
                  [
                    {
                      "letterSpacing": 1,
                      "marginHorizontal": 16,
                      "marginVertical": 9,
                      "textAlign": "center",
                    },
                    undefined,
                    {
                      "textTransform": "uppercase",
                    },
                    {
                      "color": "#6200ee",
                      "fontFamily": "System",
                      "fontWeight": "500",
                    },
                    {
                      "fontFamily": "System",
                      "fontWeight": "500",
                    },
                    {
                      "color": "#424242",
                    },
                  ],
                ]
              }
            >
              Add to cart
            </Text>
          </View>
        </View>
      </View>
      <View
        collapsable={false}
        style={
          {
            "backgroundColor": "#6200ee",
            "borderColor": "transparent",
            "borderRadius": 1,
            "borderStyle": "solid",
            "borderWidth": 0,
            "elevation": 2,
            "marginBottom": 10,
            "minWidth": 64,
            "shadowColor": "#000000",
            "shadowOffset": {
              "height": 0.75,
              "width": 0,
            },
            "shadowOpacity": 0.24,
            "shadowRadius": 1.5,
          }
        }
      >
        <View
          accessibilityRole="button"
          accessibilityState={
            {
              "disabled": true,
            }
          }
          accessible={true}
          focusable={false}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            [
              {
                "overflow": "hidden",
              },
              {
                "borderRadius": 1,
              },
            ]
          }
          testID="handleBuyItNow"
        >
          <View
            style={
              [
                {
                  "alignItems": "center",
                  "flexDirection": "row",
                  "justifyContent": "center",
                },
                undefined,
              ]
            }
          >
            <Text
              numberOfLines={1}
              selectable={false}
              style={
                [
                  {
                    "color": "#000000",
                    "fontFamily": "System",
                    "fontWeight": "400",
                  },
                  {
                    "textAlign": "left",
                  },
                  [
                    {
                      "letterSpacing": 1,
                      "marginHorizontal": 16,
                      "marginVertical": 9,
                      "textAlign": "center",
                    },
                    undefined,
                    {
                      "textTransform": "uppercase",
                    },
                    {
                      "color": "#ffffff",
                      "fontFamily": "System",
                      "fontWeight": "500",
                    },
                    {
                      "fontFamily": "System",
                      "fontWeight": "500",
                    },
                    {
                      "color": "#ffffff",
                    },
                  ],
                ]
              }
            >
              Buy it now
            </Text>
          </View>
        </View>
      </View>
    </View>
  </View>
  <View
    style={
      [
        {
          "marginVertical": 8,
        },
        {
          "marginBottom": 15,
        },
      ]
    }
    theme={
      {
        "animation": {
          "scale": 1,
        },
        "colors": {
          "accent": "#03dac4",
          "backdrop": "rgba(0, 0, 0, 0.5)",
          "background": "#f6f6f6",
          "disabled": "rgba(0, 0, 0, 0.26)",
          "error": "#B00020",
          "notification": "#f50057",
          "onSurface": "#000000",
          "placeholder": "rgba(0, 0, 0, 0.54)",
          "primary": "#6200ee",
          "surface": "#ffffff",
          "text": "#000000",
        },
        "dark": false,
        "fonts": {
          "light": {
            "fontFamily": "System",
            "fontWeight": "300",
          },
          "medium": {
            "fontFamily": "System",
            "fontWeight": "500",
          },
          "regular": {
            "fontFamily": "System",
            "fontWeight": "400",
          },
          "thin": {
            "fontFamily": "System",
            "fontWeight": "100",
          },
        },
        "roundness": 4,
      }
    }
  >
    <View>
      <View
        style={
          {
            "backgroundColor": "#f6f6f6",
          }
        }
      >
        <View
          accessibilityRole="button"
          accessibilityState={
            {
              "disabled": false,
              "expanded": false,
            }
          }
          accessible={true}
          focusable={true}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            [
              {
                "overflow": "hidden",
              },
              [
                {
                  "padding": 8,
                },
                undefined,
              ],
            ]
          }
        >
          <View
            pointerEvents="none"
            style={
              {
                "alignItems": "center",
                "flexDirection": "row",
              }
            }
          >
            <View
              style={
                [
                  {
                    "margin": 8,
                  },
                  {
                    "flex": 1,
                    "justifyContent": "center",
                  },
                ]
              }
            >
              <Text
                numberOfLines={1}
                selectable={false}
                style={
                  [
                    {
                      "color": "#000000",
                      "fontFamily": "System",
                      "fontWeight": "400",
                    },
                    {
                      "textAlign": "left",
                    },
                    [
                      {
                        "fontSize": 16,
                      },
                      {
                        "color": "rgba(0, 0, 0, 0.87)",
                      },
                      {
                        "color": "#000000",
                      },
                    ],
                  ]
                }
              >
                Description
              </Text>
            </View>
            <View
              style={
                [
                  {
                    "margin": 8,
                  },
                  undefined,
                ]
              }
            >
              <Text
                accessibilityElementsHidden={true}
                allowFontScaling={false}
                importantForAccessibility="no-hide-descendants"
                pointerEvents="none"
                selectable={false}
                style={
                  [
                    {
                      "color": "rgba(0, 0, 0, 0.87)",
                      "fontSize": 24,
                    },
                    [
                      {
                        "lineHeight": 24,
                        "transform": [
                          {
                            "scaleX": 1,
                          },
                        ],
                      },
                      {
                        "backgroundColor": "transparent",
                      },
                    ],
                    {
                      "fontFamily": "Material Design Icons",
                      "fontStyle": "normal",
                      "fontWeight": "normal",
                    },
                    {},
                  ]
                }
              >
                󰅀
              </Text>
            </View>
          </View>
        </View>
      </View>
    </View>
    <View
      style={
        [
          {
            "backgroundColor": "rgba(0, 0, 0, 0.12)",
            "height": 0.5,
          },
          undefined,
          undefined,
        ]
      }
    />
    <View>
      <View
        style={
          {
            "backgroundColor": "#f6f6f6",
          }
        }
      >
        <View
          accessibilityRole="button"
          accessibilityState={
            {
              "disabled": false,
              "expanded": false,
            }
          }
          accessible={true}
          focusable={true}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            [
              {
                "overflow": "hidden",
              },
              [
                {
                  "padding": 8,
                },
                undefined,
              ],
            ]
          }
        >
          <View
            pointerEvents="none"
            style={
              {
                "alignItems": "center",
                "flexDirection": "row",
              }
            }
          >
            <View
              style={
                [
                  {
                    "margin": 8,
                  },
                  {
                    "flex": 1,
                    "justifyContent": "center",
                  },
                ]
              }
            >
              <Text
                numberOfLines={1}
                selectable={false}
                style={
                  [
                    {
                      "color": "#000000",
                      "fontFamily": "System",
                      "fontWeight": "400",
                    },
                    {
                      "textAlign": "left",
                    },
                    [
                      {
                        "fontSize": 16,
                      },
                      {
                        "color": "rgba(0, 0, 0, 0.87)",
                      },
                      {
                        "color": "#000000",
                      },
                    ],
                  ]
                }
              >
                Shipping information
              </Text>
            </View>
            <View
              style={
                [
                  {
                    "margin": 8,
                  },
                  undefined,
                ]
              }
            >
              <Text
                accessibilityElementsHidden={true}
                allowFontScaling={false}
                importantForAccessibility="no-hide-descendants"
                pointerEvents="none"
                selectable={false}
                style={
                  [
                    {
                      "color": "rgba(0, 0, 0, 0.87)",
                      "fontSize": 24,
                    },
                    [
                      {
                        "lineHeight": 24,
                        "transform": [
                          {
                            "scaleX": 1,
                          },
                        ],
                      },
                      {
                        "backgroundColor": "transparent",
                      },
                    ],
                    {
                      "fontFamily": "Material Design Icons",
                      "fontStyle": "normal",
                      "fontWeight": "normal",
                    },
                    {},
                  ]
                }
              >
                󰅀
              </Text>
            </View>
          </View>
        </View>
      </View>
    </View>
    <View
      style={
        [
          {
            "backgroundColor": "rgba(0, 0, 0, 0.12)",
            "height": 0.5,
          },
          undefined,
          undefined,
        ]
      }
    />
  </View>
</View>
`;
